doc/neps/nep-0044-restructuring-numpy-docs.rst

```diff
@@ -0,0 +1,234 @@
+be done with the Keeling curve (decades worth of CO2 concentration in air
+measurements) rather than with synthetic random data.
+
+Ideas for tutorials (these capture the types of things that make sense, they're
+not necessarily the exact topics we propose to implement):
+
+- Conway's game of life with only NumPy (note: already in `Nicolas Rougier's book
+  <https://www.labri.fr/perso/nrougier/from-python-to-numpy/#the-game-of-life>`_)
+- Using masked arrays to deal with missing data in time series measurements
+- Using Fourier transforms to analyze the Keeling curve data, and extrapolate it.
+- Geospatial data (e.g. lat/lon/time to create maps for every year via a stacked
+  array, like `gridMet data <http://www.climatologylab.org/gridmet.html>`_)
+- Using text data and dtypes (e.g. use speeches from different people, shape
+  ``(n_speech, n_sentences, n_words)``)
+
+The *Preparing to Teach* document [2]_ from the Software Carpentry Instructor
+Training materials is a nice summary of how to write effective lesson plans (and
+tutorials would be very similar). In addition to adding new tutorials, we also
+propose a *How to write a tutorial* document, which would help users contribute
+new high-quality content to the documentation.
+
+Data sets
+---------
+
+Using interesting data in the NumPy docs requires giving all users access to
+that data, either inside NumPy or in a separate package. The former is not the
+best idea, since it's hard to do without increasing the size of NumPy
+significantly. Even for SciPy there has so far been no consensus on this (see
+`scipy PR 8707 <https://github.com/scipy/scipy/pull/8707>`_ on adding a new
+``scipy.datasets`` subpackage).
+
+So we'll aim for a new (pure Python) package, named ``numpy-datasets`` or
+``scipy-datasets`` or something similar. That package can take some lessons from
+how, e.g., scikit-learn ships data sets. Small data sets can be included in the
+repo, large data sets can be accessed via a downloader class or function.
+
+Related Work
+============
+
+Some examples of documentation organization in other projects:
+
+- `Documentation for Jupyter <https://jupyter.org/documentation>`_
+- `Documentation for Python <https://docs.python.org/3/>`_
+- `Documentation for TensorFlow <https://www.tensorflow.org/learn>`_
+
+These projects make the intended audience for each part of the documentation
+more explicit, as well as previewing some of the content in each section. 
+
+Implementation
+==============
+
+Besides rewriting the current documentation to some extent, it would be ideal
+to have a technical infrastructure that would allow more contributions from the
+community. For example, if Jupyter Notebooks could be submitted as-is as
+tutorials or How-Tos, this might create more contributors and broaden the NumPy
+community.
+
+Similarly, if people could download some of the documentation in Notebook
+format, this would certainly mean people would use less outdated material for
```

Comment:
Not sure about the meaning here; why would this mean less outdated material?

Reply:
The way I see it is that people are using notebooks provided by other parties because they can't find "official" ones (a lot of people just starting on data science or computational sciences in general would start a notebook before writing their own code).